lib/global-styles.php

```diff
@@ -295,15 +295,27 @@ function gutenberg_experimental_global_styles_get_supported_styles( $supports )
  * @return array
  */
 function gutenberg_experimental_global_styles_get_block_data() {
-	$block_data = array(
-		'global' => array(
-			'selector' => ':root',
-			'supports' => array( 'background-color' ),
-		),
-	);
+	$block_data = array();
 
 	$registry = WP_Block_Type_Registry::get_instance();
-	foreach ( $registry->get_all_registered() as $block_name => $block_type ) {
+	$blocks = array_merge(
+		$registry->get_all_registered(),
+		array(
+			'global' => new WP_Block_Type(
+				'global',
+				array(
+					'supports' => array(
+						'__experimentalSelector' => ':root',
+						'__experimentalFontSize' => true,
+						'__experimentalColor' => array(
```

Comment:
A gotcha here: the `global` context is the only one for which we define the background color but not the text color. This is possible to do if we use the post-processed syntax (what we used before), but impossible with the pre-processed syntax (like a block.json, this change).

I've prepared a follow-up PR https://github.com/WordPress/gutenberg/pull/24320/ that fixes this as well as makes the block support keys be identical to the theme.json & block attribute structure.

Reply:
What about "gradients: true" here?

Reply:
Not sure where to draw the line for what to add and what not to add in the global context, so I'm fine trying in practice -- I'll add it. Although it's probably easier to add something later than remove it :shrug:

Reply:
done at https://github.com/WordPress/gutenberg/pull/24298/commits/364252c03cb8c8d349cbd8ef7121724b28fac34d